protocol/app/app.go

```diff
@@ -1165,6 +1165,7 @@ func New(
 
 	app.ModuleManager.SetOrderPreBlockers(
```

Comment:
nit: Maybe add a comment that indicates the upgrade module's pre-blocker should always be first since it fixes up the state used by other pre-blockers?